src/components/editor/Bindings/OnIncompatibleSchemaChange/useUpdateOnIncompatibleSchemaChange.ts

```diff
@@ -0,0 +1,129 @@
+import { modifyDraftSpec } from 'api/draftSpecs';
+import { useEntityType } from 'context/EntityContext';
+import { useCallback } from 'react';
+import { useIntl } from 'react-intl';
+import { BASE_ERROR } from 'services/supabase';
+import { useBinding_bindings } from 'stores/Binding/hooks';
+import { Schema } from 'types';
+import { hasLength } from 'utils/misc-utils';
+import { getBindingIndex } from 'utils/workflow-utils';
+import {
+    useEditorStore_persistedDraftId,
+    useEditorStore_queryResponse_draftSpecs,
+    useEditorStore_queryResponse_mutate,
+} from '../../Store/hooks';
+import { AutoCompleteOption } from './types';
+
+export interface BindingMetadata {
+    bindingIndex: number;
+    collection: string;
+}
```

Comment:
This interface is also exported from `useUpdateBackfillCounter`. Would you like to consolidate the two interface definitions at this time? If not, could a TODO comment be thrown in to, at the very least, highlight the type duplication?

**ASIDE:** Curious if there should be a universal _BindingMetadata_ interface that is narrowed as needed.